src/citrine/resources/dataset.py

```diff
@@ -386,21 +412,43 @@ class DatasetCollection(Collection[Dataset]):
 
     Parameters
     ----------
-    project_id: UUID
-        Unique ID of the project this dataset collection belongs to.
+    team_id: UUID
+        Unique ID of the team this dataset collection belongs to.
     session: Session
         The Citrine session used to connect to the database.
 
     """
 
-    _path_template = 'projects/{project_id}/datasets'
     _individual_key = None
     _collection_key = None
     _resource = Dataset
 
-    def __init__(self, project_id: UUID, session: Session):
-        self.project_id = project_id
-        self.session: Session = session
+    def __init__(self,
+                 *args,
+                 session: Session = None,
+                 team_id: Optional[UUID] = None,
+                 project_id: Optional[UUID] = None):
+        # Handle positional arguments for backward compatibility
+        args = _pad_positional_args(args, 2)
+        self.project_id = project_id or args[0]
+        self.session = session or args[1]
+        if self.session is None:
+            raise TypeError("Missing one required argument: session.")
+
+        self.team_id = _data_manager_deprecation_checks(
+            session=self.session,
+            project_id=self.project_id,
+            team_id=team_id,
+            obj_type="Datasets")
+
+    # After the Data Manager deprecation
+    # this can be a Class Variable using the `teams/...` endpoint
+    @property
+    def _path_template(self):
+        if self.project_id is None:
+            return f'teams/{self.team_id}/datasets'
+        else:
+            return f'projects/{self.project_id}/datasets'
```

Comment:
If we're migrating  `_path_template` to a property, why aren't we doing that everywhere in its own PR?

Reply:
I agree that in an ideal world, this mega-PR would have been chunked up into smaller, more focused PRs. Moving everything (at least data-related) to keyword arguments is another example that comes to mind. But given it changed hands multiple times, most recently due to the layoffs, and has come under time pressure, it didn't seem like splitting it up after the fact would be worthwhile when it got to me.

Also, just to be clear, the intention isn't to migrate `_path_template` to a property everywhere. It's just a temporary approach for the endpoints relevant to data manager until the `/projects` versions are removed.

Reply:
To be clear, `_path_template` would be better written as an abstract method so that we could require every Collection to implement it, but that's definitely outside scope here.

Just giving appropriate stylistic push back.  I chose `comment` because I saw nothing blocking.